File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* deleteDuplicates(struct ListNode* head) {
    if(head==NULL || head->next==NULL)
       return head;
    struct ListNode* currentNode = head;
    while(currentNode !=NULL && currentNode->next!=NULL)
    {
          if(currentNode->val == currentNode->next->val)
          {
            struct ListNode* duplicates = currentNode->next;
            currentNode->next = currentNode->next->next;
            free(duplicates);
          }
          else
          {
            currentNode = currentNode->next;
          }
    } 
    return head;    
}

File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode c = head;
        while(c!=null && c.next!=null){
            if(c.val == c.next.val)
                c.next = c.next.next;
            else
                c=c.next;    
        }
        return head;
    }
}